supportbot/app.py

```diff
@@ -24,22 +23,71 @@ def run_app(config):
         ]
     )
     def respond_to_help_requests(message):
-        user = MeshUser(app, message['user'], config['nn_property_id'])
-
-        app.client.chat_postMessage(
-            channel=message['channel'],
-            thread_ts=message['ts'],
-            text=f"This is a reply to <@{message['user']}>! It looks like "
-                 f"your email is {user.email} {'and' if user.network_number else 'but'} your network number "
-                 f"{'is ' + user.network_number if user.network_number else 'could not be found'}. "
-                 f"Here's a diagnostics report to help our volunteers:",
-        )
-
-        upload_report_file(app, "DATA\n" * 43, message['channel'], message['ts'], user.network_number)
+        handle_support_request(app, config, message['user'], message['channel'], message['ts'])
 
     @app.event("message")
     def handle_message_events():
         """This handler silences warnings about "unhandled" message events"""
         pass
 
+    @app.shortcut("run_node_diagnostics")
+    def open_modal(ack, shortcut, client):
+        ack()
+
+        resp = client.views_open(
+            trigger_id=shortcut["trigger_id"],
+            view={
+                "type": "modal",
+                "callback_id": "manually_run_diagnostics",
+                "private_metadata": json.dumps({
+                    'channel': shortcut['channel']['id'],
+                    'ts': shortcut['message_ts'],
+                    'user': shortcut['message']['user']
+                }),
+                "title": {
+                    "type": "plain_text",
+                    "text": "Run node diagnostics?",
+                    "emoji": True
+                },
+                "submit": {
+                    "type": "plain_text",
+                    "text": "Submit",
+                    "emoji": True
+                },
+                "close": {
+                    "type": "plain_text",
+                    "text": "Cancel",
+                    "emoji": True
+                },
+                "blocks": [
+                    {
+                        "type": "section",
+                        "text": {
+                            "type": "mrkdwn",
+                            "text": "Are you sure you would like to run diagnostics for this message? This will "
```

Comment:
Do you use some type of formatter that splits up these strings?

Reply:
No I was doing it manually. But honestly this should probably just be templated and moved to its own file anyway